internal/backends/postgresql/metadata/pool/transaction.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/observability"
+)
+
+// InTransaction uses pool p and wraps the given function f in a transaction.
+//
+// If f returns an error or context is canceled, the transaction is rolled back.
+func InTransaction(ctx context.Context, p *pgxpool.Pool, f func(tx *pgx.Tx) error) (err error) {
```

Comment:
This function takes pool as parameter. Unlike sqlite, it's not using `fsql` because existing `*pgxpool.Pool` functions are quite useful.

Other option is we could create a struct similar to `fsql.DB` and wrap all existing `*pgxpool.Pool` functions like `fsql.DB` does. Then we can add our custom function like `db.InTransaction`. Do we prefer that?

Reply:
> Do we prefer that?

Let's keep it as it is now